src/silo/storage/column/pango_lineage_column.cpp

```diff
@@ -5,59 +5,54 @@ namespace silo::storage::column {
 PangoLineageColumn::PangoLineageColumn() = default;
 
 void PangoLineageColumn::insert(const common::PangoLineage& value) {
-   if (!value_id_lookup.contains(value)) {
+   if (!value_to_id_lookup.contains(value)) {
       for (const auto& parent_lineage : value.getParentLineages()) {
          initBitmapsForValue(parent_lineage);
       }
    }
-
-   const auto value_id = value_id_lookup[value];
-   indexed_values[value_id].add(sequence_count);
+   const uint32_t value_id = value_to_id_lookup[value];
+   indexed_values[value_id].add(value_ids.size());
    insertSublineageValues(value);
-   sequence_count++;
+   value_ids.push_back(value_id);
 }
 
 void PangoLineageColumn::initBitmapsForValue(const common::PangoLineage& value) {
-   if (value_id_lookup.contains(value)) {
+   if (value_to_id_lookup.contains(value)) {
       return;
    }
 
-   value_id_lookup[value] = value_id_lookup.size();
+   value_to_id_lookup[value] = value_to_id_lookup.size();
+   id_to_value_lookup.push_back(value);
    indexed_values.emplace_back();
-
-   roaring::Roaring sublineage_values;
-   for (const auto& [pango_lineage, value_id] : value_id_lookup) {
-      if (pango_lineage.isSublineageOf(value)) {
-         sublineage_values |= indexed_values[value_id];
-      }
-   }
-
-   indexed_sublineage_values.emplace_back(sublineage_values);
+   indexed_sublineage_values.emplace_back();
 }
 
 void PangoLineageColumn::insertSublineageValues(const common::PangoLineage& value) {
-   for (const auto& [pango_lineage, value_id] : value_id_lookup) {
-      if (value.isSublineageOf(pango_lineage)) {
-         indexed_sublineage_values[value_id].add(sequence_count);
-      }
+   for (const auto& pango_lineage : value.getParentLineages()) {
+      const uint32_t value_id = value_to_id_lookup[pango_lineage];
+      indexed_sublineage_values[value_id].add(value_ids.size());
    }
 }
 
 roaring::Roaring PangoLineageColumn::filter(const common::PangoLineage& value) const {
-   if (!value_id_lookup.contains(value)) {
+   if (!value_to_id_lookup.contains(value)) {
       return {};
    }
 
-   return indexed_values[value_id_lookup.at(value)];
+   return indexed_values[value_to_id_lookup.at(value)];
 }
 
 roaring::Roaring PangoLineageColumn::filterIncludingSublineages(const common::PangoLineage& value
 ) const {
-   if (!value_id_lookup.contains(value)) {
+   if (!value_to_id_lookup.contains(value)) {
       return {};
    }
 
-   return indexed_sublineage_values[value_id_lookup.at(value)];
+   return indexed_sublineage_values[value_to_id_lookup.at(value)];
 }
 
+std::string PangoLineageColumn::getAsString(std::size_t idx) const {
+   return id_to_value_lookup[value_ids[idx]].value;
+};
```

Comment:
Let's also add a small unit test for this function. The version for `IndexedStringColumn` doesn't work, but I'll fix that in my next PR.

Reply:
What do you mean that it does not work? What should the test encompass?

Reply:
The test should be:
* insert a value into the column
* get its string value by its index

Those indexed columns are relatively complicated. In the `IndexedStringColumn`, the `id_to_value_lookup` is not updated.

Reply:
Ah yes, that fix is already in #122 
Will commit that test shortly